### ISSUE TITLE
Probe Layout Builder Context to control visibility of components 

### ISSUE THREAD

**Issue by briand44** [uiowa/uiowa#6473]
# Story

As a web team, we would like to understand if the layout builder context module can safely provide control of visibility to components so that we can provide a solution to #6431.

# Additional Info

Related issue https://github.com/uiowa/uiowa/issues/6431

https://github.com/uiowa/uiowa/issues/6431#issuecomment-1560010325

# Proposed solution

**Comment by bspeare** [comment]
# Drupal core block visibility settings

- Tried out Drupal 10 dev with https://www.drupal.org/project/drupal/issues/2916876 with D10 patch (https://www.drupal.org/files/issues/2023-03-10/2916876-167.10x.patch) https://www.drupal.org/project/drupal/issues/2916876#comment-14962025 on simplytest.me.  
  - Works great and seems like better UX than Layout Builder Context because everything is set within the block. 
    - Separate edit tray makes settings these values better UX
    - Unclear if query parameters can be set here, I'm guessing not. 
      - Maybe once this is released there will be more integration? 
    - Request path functionality here seems intended for something like:
      - Only show an apply now button on areas of study pages based on taxonomy terms or url paths. 
    - **No caching issues.**  

# Layout Builder Context
- Tried out https://www.drupal.org/project/layout_builder_context
  - Similar functionality to https://www.drupal.org/project/drupal/issues/2916876
  -  With this module, you have to set a context on the context module page `/admin/structure/context` and **then** set it on each block which might not be the most user-friendly option for webmasters. 
  - Settings are lumped into the block edit screen tray and can make this tray overcrowded.
  - The `request_parameter_condition` module works with `layout_builder_context`, but only worked after I cleared the site cache each time. 
    - https://www.drupal.org/project/request_parameter_condition - supported
      - These modules do the same but don't look very active:
        - https://www.drupal.org/project/condition_query 
        - https://www.drupal.org/project/request_data_conditions
   - **Caching issues on simplytest.me and local ddev testing.**  

Long term, my hope is that https://www.drupal.org/project/drupal/issues/2916876 will work with modules like https://www.drupal.org/project/request_parameter_condition when it is released.

**Comment by pyrello** [comment]
Just noting that, if we find a module that adds a condition plugin for reacting to query parameters without much other overhead, I'm cool with that. However, we can also roll our own if a contrib module is adding more code overhead than we want.

**Comment by joewhitsitt** [comment]
Ben had demonstrated the two solutions above to Brian and I on Friday. I was thinking that the Drupal 10 solution looks to be more inline with what we would want and that https://www.drupal.org/project/condition_query or https://www.drupal.org/project/request_parameter_condition would probably be visible within the D10 solution once the module is compatible since core is just making that block functionality visible in the lb settings tray.

It would be nice to confirm that with a simplytestme or other test environment with the necessary patches, etc.

**Comment by cory-skeers** [comment]
Tested a bit with both drupal core 10.x visibility settings on simplytest.me and layout builder context locally, and came up with pretty similar conclusions.

I struggled to get LBC to really work, and even when it did, ran into the same caching issues. With a little work on the caching stuff, it seems like it'd be really well-suited for something like role-based customizations on an intranet site, where there can be a small handful of contexts defined for all pages on a site. For one-off campaign focuses, or something like that, it could get unwieldy fast. I think we could make it work, but that it'd take a lot of overhead to basically make it work like the D10 block visibility work is aiming to do.

The D10 core visibility settings look great. I don't think they'll cover what we're looking for out of the box, but it looks pretty easy to tie into and use, where we could build a lightweight, homegrown option if the community doesn't end up providing a full-fledged solution by then .

If there were a push to have something like this available in the nearer term, I also think we could build a homegrown version of this functionality with the aim of swapping it out for core in the future; with the downfalls of creating work intended to be dropped in the future (and a few potential roadblocks in getting our custom stuff visible in third-party block settings everywhere it's needed, as opposed to the ease of using core settings).
